### ISSUE TITLE
Regent: Casts on futures cause blocking

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#417]
If you -fpretty 1 the following code:

```
import 'regent'
task foo()
  return 12
end
task bar(x : double)
  return x + 1.0
end
task main()
  var x = foo()
  var y = bar(x)
end
```

you will see that the compiler handles the (implicit) cast on the future for x by emitting a blocking call to `__future_get_result`:

```
task main()
  var $x : future(int32) = foo()
  var $y : future(double) = bar(double(__future_get_result($x)))
end
```

There could be a warning for this behavior or, even better, the cast could be lifted into its own task, as happens already for arithmetic operations on futures.

**Comment by elliottslaughter** [comment]
Should be fixed by 7cbe03ea1babf1737b1b487af1fae1b173573d15.

**Comment by manopapad** [comment]
Fixed!
